src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -165,20 +76,20 @@ private void TryOpeningDocumentsForNewCookie(uint cookie)
                     }
                     else
                     {
-                        activeContextProjectId = GetActiveContextProjectIdAndWatchHierarchies(cookie, documentIds.Select(d => d.ProjectId));
+                        activeContextProjectId = GetActiveContextProjectId(cookie, documentIds.Select(d => d.ProjectId));
                     }
 
-                    if ((object)_runningDocumentTable.GetDocumentData(cookie) is IVsTextBuffer bufferAdapter)
+                    if ((object)RunningDocumentTable.GetDocumentData(cookie) is IVsTextBuffer bufferAdapter)
                     {
-                        var textBuffer = _editorAdaptersFactoryService.GetDocumentBuffer(bufferAdapter);
+                        var textBuffer = EditorAdaptersFactoryService.GetDocumentBuffer(bufferAdapter);
 
                         if (textBuffer != null)
                         {
                             var textContainer = textBuffer.AsTextContainer();
 
                             foreach (var documentId in documentIds)
                             {
-                                if (!w.IsDocumentOpen(documentId) && !_workspace._documentsNotFromFiles.Contains(documentId))
+                                if (!w.IsDocumentOpen(documentId) && IsDocumentTracked(documentId))
                                 {
                                     var isCurrentContext = documentId.ProjectId == activeContextProjectId;
                                     if (w.CurrentSolution.ContainsDocument(documentId))
```

Comment:
(referring to the earlier comment) -- start what is the derived behavior *here* -- keep the rest still in the base.